app/src/common/shared/org/mozilla/vrbrowser/telemetry/GleanMetricsService.java

```diff
@@ -53,6 +69,100 @@ public static void stop() {
         Glean.INSTANCE.setUploadEnabled(false);
     }
 
+    public static void startPageLoadTime() {
+       // pageLoadingTimerId = Pages.INSTANCE.getPageLoad().start();
+    }
+
+    public static void stopPageLoadTimeWithURI(String uri) {
+      //  Pages.INSTANCE.getPageLoad().stopAndAccumulate(pageLoadingTimerId);
+
+        try {
+            URI uriLink = URI.create(uri);
+            if (uriLink.getHost() == null) {
+                return;
+            }
+
+            if (domainMap.add(UrlUtils.stripCommonSubdomains(uriLink.getHost()))) {
+                Url.INSTANCE.getDomains().add();
+            }
+            Url.INSTANCE.getVisits().add();
+
+        } catch (IllegalArgumentException e) {
+            Log.e(LOGTAG, "Invalid URL", e);
+        }
+
+    }
+
+    public static void sessionStop() {
+        domainMap.clear();
+
+        // TODO: Because we measure it by sessions,
+        // How to reset
+        // Url.INSTANCE.getDomains() and Url.INSTANCE.getVisits() counter to zero?
+    }
```

Comment:
@Dexterp37 . We have some metrics are measured by sessions, so we have to reset the `Counter` to zero after a session. Is there a way to do this?

Reply:
What is a `session`? There is no such thing as a `session` concept in the Glean world.

Metrics are automatically cleared depending on their [lifetime](https://mozilla.github.io/glean/book/user/metric-parameters.html#optional-metric-parameters).

The `metrics` ping, for example, is sent once per day at 4am (or other times, if it's overdue). If a metric has `lifetime: ping` (default), then its value is cleared automatically after the ping is assembled.

This means that you don't really need to care about this, unless you really want to measure things in a different cadence.

For the specific `domains` metric, since you want to measure the number of unique `domains` visited, this could be trickier. @chutten, thoughts?

Reply:
One session means the duration of this application is running at the foreground. So, while the application switch to the background, our application will receive an `onStop()` event, and that means the current session has been ended. The session lifetime sounds like very close to `lifetime: application`? if it is not totally the same, I guess I still can use `lifetime: ping` to replace it.

Reply:
I think I can use custom pings, `session_end`, to help me send these metrics once application jumps to the background.